packages/react-native-sdk/src/components/NetworkQualityIndicator.tsx

```diff
@@ -0,0 +1,72 @@
+import { StyleSheet, View } from 'react-native';
+import { Path, Svg } from 'react-native-svg';
+import { theme } from '../theme';
+import { SfuModels } from '@stream-io/video-client';
+
+/**
+ * Props to be passed for the NetworkQualityIndicator component.
+ */
+export type NetworkQualityIndicatorType = {
+  connectionQuality: SfuModels.ConnectionQuality;
+};
+
+export const NetworkQualityIndicator = ({
+  connectionQuality,
+}: NetworkQualityIndicatorType) => {
+  if (!connectionQuality) return null;
+
+  const connectionQualitySignalColors: Record<
+    SfuModels.ConnectionQuality,
+    string[]
+  > = {
+    0: [
+      theme.light.static_white,
+      theme.light.static_white,
+      theme.light.static_white,
+    ],
+    1: [theme.light.error, theme.light.static_white, theme.light.static_white],
+    2: [theme.light.primary, theme.light.primary, theme.light.static_white],
+    3: [theme.light.primary, theme.light.primary, theme.light.primary],
+  };
+
+  const connectionQualityColors =
+    connectionQualitySignalColors[connectionQuality];
+
+  return (
+    <View style={[styles.container, theme.icon.lg]}>
```

Comment:
A minor suggestion: 

```
StyleSheet.create({
  container: {
    backgroundColor: theme.light.static_overlay,
    borderRadius: theme.rounded.xs,
   ...theme.icon.lg
  },
});
```

Better to do the above. Than to create an array unnecessarily here.

Reply:
Umm, since the type of theme.icon.lg is StyleProp<ViewStyle>, you cannot spread it. 

Getting:
```
Spread types may only be created from object types.ts(2698)
(property) lg: StyleProp<ViewStyle>
```